Source/ORTS.Common/VirtualFileSystem.cs

```diff
@@ -0,0 +1,717 @@
+// 
+// You should have received a copy of the GNU General Public License
+// along with Open Rails.  If not, see <http://www.gnu.org/licenses/>.
+
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Diagnostics;
+using System.Text.RegularExpressions;
+using System.IO;
+using System.IO.Compression;
+using SharpCompress.Archives;
+using SharpCompress.Readers;
+
+namespace ORTS.Common
+{
+    /// <summary>
+    /// Virtual File System. Can be <see cref="Initialize(string)"/>'d from a config file with lines e.g. C:\dir\archive.zip\subdir\ /MSTS/mountpoint/,
+    /// or from a directory. Found subsequent archives are auto-mounted to their respective locations.
+    /// <br><see cref="Path.Combine"/> like functions continue to work on virtual paths, as well as both the "\" and "/" separators.
+    /// But <see cref="Path.GetFullPath"/> must never be used on them, because that adds an unwanted drive letter.</br>
+    /// <br><see cref="File.Exists"/> becomes <see cref="Vfs.FileExists"/>, <see cref="Directory.Exists"/> becomes <see cref="Vfs.DirectoryExists"/>,
+    /// <see cref="Directory.GetFiles(string, string)"/> becomes <see cref="Vfs.GetFiles"/>.</br>
+    /// </summary>
+    public static class Vfs
+    {
+        static VfsNode VfsRoot;
+        static readonly Stack<(string, VfsNode)> Stack = new Stack<(string, VfsNode)>();
+        static readonly ConcurrentDictionary<string, object> OpenArchives = new ConcurrentDictionary<string, object>();
+        static readonly List<string> SupportedArchiveExtensions = new List<string> { ".zip", ".rar" };
+        internal static readonly ConcurrentQueue<string> InitLog = new ConcurrentQueue<string>();
+
+        ///////////////////////////////////////////////////////////////////////////////////////
+        ///
+        /// The configuration file consits of lines of format:
+        /// Drive:\SystemPath\ /MountPoint/
+        ///
+        /// SystemPath - may use "/" or "\", but MountPoint may use "/" only as a directory separator.
+        ///            - with spaces in it must be quoted, e.g. "C:\My Routes\USA85\" /MSTS/ROUTES/USA85/
+        ///            - referring to a directory must end with "/" or "\" to assure that only _below_ dirs and files will be visible _below_ the mount point.
+        ///            - referring to an archive must not end with "/" or "\", e.g. C:\TEMP\MSTS1.2.zip /MSTS/
+        ///            - may refer to an archive subdirectory, e.g.: C:\routes.zip\USA3\ /MSTS/ROUTES/USA3/
+        ///            - may _not_ refer to a single non-archive file, neither a non-archive file within an archive.
+        ///            - is case-insensitive.
+        /// MountPoint - must end with "/" to avoid confusion and to assure that mounting will be done _below_ the given path.
+        ///            - may not contain spaces, may not be quoted.
```

Comment:
There are many problems with using a plain text file for this (and a file at all), but this line is particularly egregious to me. Why can't we have spaces in the mounted paths?

Reply:
Nothing special, just didn't want to further complicate the parsing regex, and it seemed not be inevitable to have spaces there.

Reply:
I would argue that it is always inevitable to have spaces in a path/filename :)